Added missing properties and methods to various ol.source.* classes.

gkresic opened the pull request.

TDesjardins commented:
Nice! But please split the pullrequests by feature next time. For example the VectorTile should have it's own pr. In version history it is then easier to see id and when a new feature was added.

gkresic commented:
Sorry about that. `VectorTile` wasn't even planned at this point, I added it by inertia when was going through class hierarchy in OL and checking property by property.

Should have commit that in separate branch. Still learning...

TDesjardins commented:
;-) Thanks!

gkresic commented:
It looks like I duplicated things around `tileload*` events.

`ol.source.Tile` already had `Event` class defined (although just a placeholder), so I populated it with `getTile()` property and added methods for adding tile load listeners (`addTileLoad*Listener`).

Now I see that there are `ol.event.TileLoad*Listener`s and `OLUtil` has `addTileLoad*Listener` methods (although, they operate on `UrlTile`, not `Tile`).

Subjective as it is, to me it looks that mine implementation is a bit cleaner (since it is the same event, there is not need for separate listener interfaces, and static methods are less readable than member methods on proper objects), so I assume following can be marked as @Deprecated and removed at some point:

```
ol.event.TileLoadEndListener
ol.event.TileLoadErrorListener
ol.event.TileLoadStartListener
ol.OLUtil.addTileLoadErrorListener(UrlTile, TileLoadErrorListener)
ol.OLUtil.addTileLoadStartListener(UrlTile, TileLoadStartListener)
ol.OLUtil.addTileLoadEndListener(UrlTile, TileLoadEndListener)
```

TDesjardins commented:
Yes, right! Could you pullrequest this?
